File: practice/src/org/acme/grabber/pages/TensionCalcPage.java
package org.acme.grabber.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;

import static selenium.WebDriverHolder.getDriver;

public class TensionCalcPage {

    public TensionCalcPage setStringSizeByText(String stringSize) {
        getStringSizeSelect().selectByVisibleText(stringSize);
        return this;
    }

    public TensionCalcPage setNoteSizeByText(String noteValue) {
        getNoteSizeSelect().selectByVisibleText(noteValue);
        return this;
    }

    public TensionCalcPage clickCalculateButton() {
        getCalculateButton().click();
        return this;
    }

    public String getCalculationOutputText() {
        return getDriver().findElement(By.cssSelector("#output")).getText();
    }

    private Select getStringSizeSelect() {
        return new Select(getDriver().findElement(By.cssSelector("#string_size")));
    }

    private Select getNoteSizeSelect() {
        return new Select(getDriver().findElement(By.cssSelector("#the_note")));
    }

    private WebElement getCalculateButton() {
        return getDriver().findElement(By.cssSelector("input[value='Calculate']"));
    }

}

File: practice/src/org/acme/grabber/pages/TensionCalcPage.kt
package org.acme.grabber.pages

import org.acme.selenium.WebDriverHolder.webDriver
import org.acme.selenium.elementByCss
import org.openqa.selenium.support.ui.Select

object TensionCalcPage {
    const val url: String = "https://evertune.com/faq/resources/string_tension_calculator.php"

    private fun noteSizeSelect() = Select(webDriver.elementByCss("#the_note"))
    private fun calculateButton() = webDriver.elementByCss("input[value='Calculate']")
    private fun stringSizeSelect() = Select(webDriver.elementByCss("#string_size"))

    fun getCalculationOutputText() = webDriver.elementByCss("#output").text ?: ""

    fun setStringSize(stringSize: String) = apply {
        stringSizeSelect().selectByVisibleText(stringSize)
    }

    fun setNoteSize(noteValue: String) = apply {
        noteSizeSelect().selectByVisibleText(noteValue)
    }

    fun clickCalculateButton() = apply {
        calculateButton().click()
    }
}